src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.TokenExpiration
+{
+    internal class TokenExpirationService : BackgroundService, IDisposable
+    {
+        private readonly TaskCompletionSource<object> _standby = new();
+        private readonly IEnvironment _environment;
+        private readonly ILogger<TokenExpirationService> _logger;
+        private IDisposable _listener;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _analysisScheduled;
+
+        public TokenExpirationService(IEnvironment environment, ILogger<TokenExpirationService> logger, IOptionsMonitor<StandbyOptions> standbyOptionsMonitor)
+        {
+            _environment = environment;
+            _logger = logger;
+            if (standbyOptionsMonitor.CurrentValue.InStandbyMode)
+            {
+                _listener = standbyOptionsMonitor.OnChange(standbyOptions =>
+                {
+                    if (!standbyOptions.InStandbyMode && !_analysisScheduled)
+                    {
+                        _standby.TrySetResult(null);
+                    }
+                });
+            }
+            else
+            {
+                _standby.TrySetResult(null);
+            }
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            await Task.Yield(); // force a yield, so our synchronous code does not slowdown startup
+            await _standby.Task.WaitAsync(cancellationToken);
+
+            if (_environment.IsCoreTools())
+            {
+                return;
+            }
+
+            // Execute first during startup
+            ScheduleTokenExpirationCheck();
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                // check at every hour
+                await Task.Delay(TimeSpan.FromHours(1), cancellationToken)
+               .ContinueWith(t => ScheduleTokenExpirationCheck());
+            }
+
+            // Dispose of the listener for the standby options monitor
+            _listener?.Dispose();
```

Comment:
Move this to the `Dispose` implementation. 

`BackgroundService` already implements `IDisposable`, so make sure your implementation contains a `base.Dispose();` call.